### PR TITLE
Indicator forms in React

### PR DESCRIPTION
**Pull request by Creaticide**
Forms for adding/editing user indicators.

### CODE REVIEW

File: bundles/statistics/statsgrid/handler/MyIndicatorsHandler.js [hunk 1]
```diff
@@ -79,13 +80,11 @@ class IndicatorsHandler extends StateHandler {
     }
 
     addNewIndicator () {
-        const formFlyout = this.instance.getFlyoutManager().getFlyout('indicatorForm');
-        formFlyout.showForm(this.userDsId);
+        this.stateHandler.getController()?.getFormHandler()?.getController()?.showIndicatorPopup(this.userDsId);
```
[the remark this answers is not in the document]
Comment (ZakarFin) [reply bundles/statistics/statsgrid/handler/MyIndicatorsHandler.js:83]: Is `FormHandler` the `SearchHandler` or something else?

[the remark this answers is not in the document]
Comment (ZakarFin) [reply bundles/statistics/statsgrid/handler/MyIndicatorsHandler.js:83]: At least the stateHandler.getController() part seems unnecessary here:
```suggestion
        this.stateHandler.getFormHandler()?.getController()?.showIndicatorPopup(this.userDsId);
```

[the remark this answers is not in the document]
Comment (ZakarFin) [reply bundles/statistics/statsgrid/handler/MyIndicatorsHandler.js:83]: Maybe event like this:
```suggestion
        this.stateHandler.getFormHandler()?.showIndicatorPopup(this.userDsId);
```
Is there a possibility that getFormHandler isn't available? It is created on the contructor of StatisticsHandler.

[the remark this answers is not in the document]
Comment (Creaticide) [reply bundles/statistics/statsgrid/handler/MyIndicatorsHandler.js]: Now it's like this:
`this.fromHandler.getController().showIndicatorPopup(this.userDsId);`
Form handler should be always available.

[the remark this answers is not in the document]
Comment (ZakarFin) [reply bundles/statistics/statsgrid/handler/MyIndicatorsHandler.js]: The getController() isn't needed here, it's mostly for the React-components for restricting access to getters, but it's ok for now :)

File: bundles/statistics/statsgrid/handler/IndicatorFormHandler.js [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+    }
+
+    onEvent (e) {
+        var handler = this.eventHandlers[e.getName()];
+        if (!handler) {
+            return;
+        }
+
+        return handler.apply(this, [e]);
+    }
```
Comment (ZakarFin) [inline bundles/statistics/statsgrid/handler/IndicatorFormHandler.js:511]: We can remove these if there are no events to listen to.

File: bundles/statistics/statsgrid/handler/IndicatorFormHandler.js [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+        const promise = new Promise((resolve, reject) => {
+            this.service.getRegions(regionset.id, (err, regions) => {
+                if (err) {
+                    reject(new Error(Oskari.getMsg('StatsGrid', 'errors.regionsDataError')));
+                    return;
+                }
+                const showDataForm = (regions, data) => {
+                    data = data || {};
+                    formRegions = [...regions].sort((a, b) => a.name.localeCompare(b.name)).map((region) => {
+                        // TODO: include existing values per region when editing existing dataset
```
Comment (ZakarFin) [inline bundles/statistics/statsgrid/handler/IndicatorFormHandler.js:231]: Is there something TODO/missing here still? Looks like showDataForm() is called with the data if we have an indicator id?

Comment (Creaticide) [reply bundles/statistics/statsgrid/handler/IndicatorFormHandler.js]: Should be good to to go, TODO came with copy-paste from old code.

**Pull request merged**